pkg/agent/nodeportlocal/npl_agent_test.go

```diff
@@ -389,24 +412,29 @@ func TestPodDelete(t *testing.T) {
 }
 
 // TestPodAddMultiPort creates a Pod with multiple ports and verifies that the Pod's NPL annotation
-// and the local port table are updated correctly.
+// and the local port table are updated with only one port used as target port of the Service.
 func TestPodAddMultiPort(t *testing.T) {
```

Comment:
I think we're back to my original comment. Why are we modifying this test like this, instead of having 2 separate tests
- TestServiceMultipleTargetPorts: Service has 2 target ports pointing to 2 different container ports, both are exposed with NPL
- TestPodContainerPortWithoutService: a Pod has 2 container ports, only one of them exposed with NPL (this is your current test, but I feel like the name is confusing)

Or do we already have a test for the first case?